suite-native/module-trading/src/selectors/commonSelectors.ts

```diff
@@ -1,4 +1,32 @@
+import {
+    Feature,
+    MessageSystemRootState,
+    selectIsFeatureEnabled,
+} from '@suite-common/message-system';
+import {
+    FeatureFlag,
+    FeatureFlagsRootState,
+    selectIsFeatureFlagEnabled,
+} from '@suite-native/feature-flags';
+
 import { TradingRootState } from '../tradingSlice';
 
 export const selectTradingEnvironment = (state: TradingRootState) =>
     state.wallet.tradingNew.tradingEnvironment;
+
+export const selectIsTradingBuyEnabled = (state: MessageSystemRootState & FeatureFlagsRootState) =>
+    selectIsFeatureFlagEnabled(state, FeatureFlag.IsTradingBuyEnabled) ||
+    selectIsFeatureEnabled(state, Feature.trading.buy, false);
+
+export const selectIsTradingSwapEnabled = (state: MessageSystemRootState & FeatureFlagsRootState) =>
+    selectIsFeatureFlagEnabled(state, FeatureFlag.IsTradingSwapEnabled) ||
+    selectIsFeatureEnabled(state, Feature.trading.swap, false);
+
+export const selectIsTradingSellEnabled = (state: MessageSystemRootState & FeatureFlagsRootState) =>
+    selectIsFeatureFlagEnabled(state, FeatureFlag.IsTradingSellEnabled) ||
+    selectIsFeatureEnabled(state, Feature.trading.sell, false);
+
+export const selectIsTradingEnabled = (state: MessageSystemRootState & FeatureFlagsRootState) =>
```

Comment:
There is no way to keep the Trading tab visible, but do not show the content of it except the context message explaining why it's disabled, but I guess that is fine. Global banner can be used if really needed.